cpmpy/expressions/python_builtins.py

```diff
@@ -70,14 +71,14 @@ def any(iterable):
         return collect[0]
     if len(collect) >= 2:
         return Operator("or", collect)
-    return False
+    return BoolVal(False)
 
 def max(iterable):
     """
         max() overwrites python built-in,
         checks if all constants and computes np.max() in that case
     """
-    if not any(isinstance(elem, Expression) for elem in iterable):
+    if is_false_cst(any(isinstance(elem, Expression) for elem in iterable)):
```

Comment:
this already shows that overwriting 'any' is not a good idea...

Note that I now implemented `__bool__()` for BoolVal, so the original code would work now (with not), but still, better without for now